src/main/java/appeng/client/gui/implementations/GuiFluidTerminal.java

```diff
@@ -0,0 +1,362 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.client.gui.implementations;
+
+
+import java.io.IOException;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.lwjgl.input.Mouse;
+
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.inventory.ClickType;
+import net.minecraft.inventory.Slot;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fml.common.Loader;
+
+import appeng.api.config.Settings;
+import appeng.api.storage.data.IAEFluidStack;
+import appeng.api.util.IConfigManager;
+import appeng.api.util.IConfigurableObject;
+import appeng.client.gui.AEBaseMEGui;
+import appeng.client.gui.widgets.GuiImgButton;
+import appeng.client.gui.widgets.GuiScrollbar;
+import appeng.client.gui.widgets.ISortSource;
+import appeng.client.gui.widgets.MEGuiTextField;
+import appeng.client.me.FluidRepo;
+import appeng.client.me.SlotFluidME;
+import appeng.client.me.InternalFluidSlotME;
+import appeng.container.implementations.ContainerFluidTerminal;
+import appeng.container.slot.ISlotFluid;
+import appeng.core.AELog;
+import appeng.core.localization.GuiText;
+import appeng.core.sync.network.NetworkHandler;
+import appeng.core.sync.packets.PacketInventoryAction;
+import appeng.core.sync.packets.PacketValueConfig;
+import appeng.helpers.InventoryAction;
+import appeng.parts.reporting.PartFluidTerminal;
+import appeng.util.IConfigManagerHost;
+import appeng.util.Platform;
+import appeng.util.item.AEFluidStack;
+
+
+/**
+ * @author BrockWS
+ * @version rv6 - 12/05/2018
+ * @since rv6 12/05/2018
+ */
+public class GuiFluidTerminal extends AEBaseMEGui implements ISortSource, IConfigManagerHost
+{
+	private final List<SlotFluidME> meFluidSlots = new LinkedList<>();
+	private final FluidRepo repo;
+	private final IConfigManager configSrc;
+	private final ContainerFluidTerminal container;
+	private final int offsetX = 9;
+	private int rows = 6;
+	private int maxRows = Integer.MAX_VALUE;
+	private int perRow = 9;
+
+	protected PartFluidTerminal terminal;
+
+	private MEGuiTextField searchField;
+	private GuiImgButton sortByBox;
+	private GuiImgButton sortDirBox;
+
+	public GuiFluidTerminal( InventoryPlayer inventoryPlayer, PartFluidTerminal terminal )
+	{
+		super( new ContainerFluidTerminal( inventoryPlayer, terminal ) );
+		this.terminal = terminal;
+		this.xSize = 185;
+		this.ySize = 222;
+		final GuiScrollbar scrollbar = new GuiScrollbar();
+		this.setScrollBar( scrollbar );
+		this.repo = new FluidRepo( scrollbar, this );
+		this.configSrc = ( (IConfigurableObject) this.inventorySlots ).getConfigManager();
+		( this.container = (ContainerFluidTerminal) this.inventorySlots ).setGui( this );

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment out of this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)